packages/server/modules/workspaces/domain/operations.ts

```diff
@@ -316,3 +316,12 @@ export type UpdateWorkspaceJoinRequestStatus = (params: {
   userId: string
   status: WorkspaceJoinRequestStatus
 }) => Promise<Pick<WorkspaceJoinRequest, 'workspaceId' | 'userId'> | undefined>
+
+export type CreateWorkspaceJoinRequest = (
+  params: Omit<WorkspaceJoinRequest, 'createdAt' | 'updatedAt'>
```

Comment:
What if you need to add extra params in the future? You'll either have to introduce a 2nd positional org or pollute the WorkspaceJoinRequest obj w/ un related props - both aren't great options.

I would put the join request under a nested key like `request: WorkspaceJoinRequest` in the params obj